common/metrics.py

```
@@ -0,0 +1,209 @@
+            cls._instance.cached_queryset_miss = cls.cached_queryset_miss
+            cls._instance.cached_queryset_evicted = cls.cached_queryset_evicted
+            cls._instance.cached_queryset_invalidated = cls.cached_queryset_invalidated
+            cls._instance.cache_module_loaded = cls.cache_module_loaded
+            cls._instance.cache_choices_count = cls.cache_choices_count
+            cls._instance.cache_redis_keys_count = cls.cache_redis_keys_count
+            cls._instance.cache_operation_duration = cls.cache_operation_duration
+            cls._instance.database_query_duration = cls.database_query_duration
+        return cls._instance
+
+    @classmethod
+    def increment_cache(cls, model: str, cache_type: str, reason: str = None) -> None:
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)